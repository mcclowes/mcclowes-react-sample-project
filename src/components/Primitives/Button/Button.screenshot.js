import Button from "./index";
import Icon from "../Icon";
import React from "react";

import { TestScreenshots } from "../../../utils/testing/testScreenshots";

TestScreenshots({
  name: "Button",
  desktopOverrides: { disabled: true },
  components: [
    {
      name: "default",
      component: (
        <Button>
          <Icon icon="TICK" />
          Hello World
        </Button>
      ),
    },
  ],
});
