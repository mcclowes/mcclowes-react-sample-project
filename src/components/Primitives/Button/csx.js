import styled from "styled-components/macro";

export const ButtonWrapper = styled.button`
  align-items: center;
  border-radius: ${(props) => props.theme.sizes.border.radius};
  border: 2px solid ${(props) => props.theme.colors.primary};
  cursor: pointer;
  display: flex;
  font-family: "Cantarell", sans-serif;
  font-size: 14px;
  justify-items: center;
  padding: 0.5em 1em;

  ${(props) =>
    props.subtle
      ? `
      background: ${props.theme.colors.white};
      color: ${props.theme.colors.primary};
    `
      : `
      background: ${props.theme.colors.primary};
      color: ${props.theme.colors.white};
    `}

  &:hover {
    opacity: 0.75;
  }

  &:active {
    opacity: 0.5;
  }

  &:focus {
    outline: none;
    border: 2px solid ${(props) => props.theme.colors.alt};
  }

  ${(props) => (props.fill ? "width: 100%;" : "")};

  ${(props) => (props.large ? `padding: 0.75em 1.5em;` : "")};

  svg {
    margin-right: 0.5em;
    fill: ${(props) => props.theme.colors.white};
  }
`;

ButtonWrapper.defaultProps = {
  theme: {
    colors: {
      primary: "#aaaaaa",
      alt: "#888888",
      white: "#ffffff",
    },
    sizes: {
      border: {
        radius: "3px",
      },
    },
  },
  subtle: false,
};
