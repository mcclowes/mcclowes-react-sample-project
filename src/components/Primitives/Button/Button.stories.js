import Button from "./Button";
import Icon from "../Icon";
import React from "react";

export default { title: "Button" };

export const empty = () => {
  return <Button />;
};

export const withText = () => {
  return <Button>Hello World</Button>;
};

export const withTextProp = () => {
  return <Button text="Hello World" />;
};

export const withIcon = () => {
  return (
    <Button>
      <Icon icon="TICK" />
      Hello World
    </Button>
  );
};

export const subtle = () => {
  return <Button subtle={true}>Hello World</Button>;
};
