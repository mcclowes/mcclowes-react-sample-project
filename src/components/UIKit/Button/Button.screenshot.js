import Button from "./index";
import Icon from "../Icon";
import React from "react";

import { TestScreenshots } from "../../../utils/testing/testScreenshots";

TestScreenshots({
  name: "Button",
  components: [
    {
      name: "default",
      component: (
        <Button>
          <Icon icon="TICK" />
          Hello World
        </Button>
      ),
    },
  ],
});
