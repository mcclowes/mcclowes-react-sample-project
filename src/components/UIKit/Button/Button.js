import PropTypes from "prop-types";
import React from "react";
import { ButtonWrapper } from "./csx";

const Button = (props) => {
  const { children, text, to = "/", doClick, subtle } = props;

  return (
    <ButtonWrapper onClick={doClick} to={to} subtle={subtle}>
      {text}
      {children}
    </ButtonWrapper>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  doClick: PropTypes.func,
  text: PropTypes.string,
  to: PropTypes.string,
};

export default Button;
