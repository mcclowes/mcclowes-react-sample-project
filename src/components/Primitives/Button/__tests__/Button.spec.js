import Button from "../Button";
import React from "react";
import { shallow } from "enzyme";

let wrapper;

describe("Button", () => {
  beforeEach(() => {
    wrapper = shallow(<Button text="Default Button" />);
  });

  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("to", () => {
    it("default", () => {
      wrapper = shallow(<Button text="Default Button" />);

      expect(wrapper.dive().prop("to")).toBe("/");
    });

    it("passed", () => {
      wrapper = shallow(<Button text="Default Button" to="/some/url" />);

      expect(wrapper.dive().prop("to")).toBe("/some/url");
    });
  });
});
