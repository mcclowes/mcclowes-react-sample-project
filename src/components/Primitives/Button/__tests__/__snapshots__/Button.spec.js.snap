// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component 1`] = `
<csx__ButtonWrapper
  subtle={false}
  theme={
    Object {
      "colors": Object {
        "alt": "#888888",
        "primary": "#aaaaaa",
        "white": "#ffffff",
      },
      "sizes": Object {
        "border": Object {
          "radius": "3px",
        },
      },
    }
  }
  to="/"
>
  Default Button
</csx__ButtonWrapper>
`;
